### DISCUSSION TITLE
Does the new pico w bluetooth support not have all of the config options?

### DISCUSSION THREAD

**Discussion by matteius** [micropython/micropython#11793] [Using MicroPython] [answered]
Hey -- I installed `micropython-firmware-pico-w-130623.uf2` and also tried `rp2-pico-w-20230615-unstable-v1.20.0-219-g47dc7d013.uf2` and I've been following the documentation examples:  https://github.com/micropython/micropython/blob/master/examples/bluetooth/ble_bonding_peripheral.py

It seems that part of the example is passing config values which are not recognized by these micropython firmwares.
```
class BLETemperature:
    def __init__(self, ble, name="mpy-temp"):
        self._ble = ble
        self._load_secrets()
        self._ble.irq(self._irq)
        self._ble.config(bond=True)
        self._ble.config(le_secure=True)
        self._ble.config(mitm=True)
        self._ble.config(io=_IO_CAPABILITY_DISPLAY_YESNO)
        self._ble.active(True)
        self._ble.config(addr_mode=2)
        ((self._handle,),) = self._ble.gatts_register_services((_ENV_SENSE_SERVICE,))
        self._connections = set()
        self._payload = advertising_payload(
            name=name, services=[_ENV_SENSE_UUID], appearance=_ADV_APPEARANCE_GENERIC_THERMOMETER
        )
        self._advertise()
```

Specifically had to comment out all of these config values to run:
```
        self._ble.config(bond=True)
        self._ble.config(le_secure=True)
        self._ble.config(mitm=True)
        self._ble.config(io=_IO_CAPABILITY_DISPLAY_YESNO)
        self._ble.config(addr_mode=2)
```

The bluetooth is discoverable by my android phone, but I am unable to pair with it and I suspect it is because these config values are actually needed.   Can anyone advise?

**Comment by jimmo** [comment, marked as answer]
@matteius Pairing & bonding is not currently enabled (it's controlled via the `MICROPY_PY_BLUETOOTH_ENABLE_PAIRING_BONDING` option). It's partially implemented for btstack (which is the host stack we use on the Pico W). We hope to have this working feature-parity with the NimBLE-based ports by the next release (v1.21, targeting end of this month).

**Reply by ZacharyInDallas** [reply]
I ran into the same problem using the keyboard example.  I've seen the release notes for the full release, but haven't seen any release notes for the nightly builds.  Are their any notes on the release builds that mention the limitations / bugs with a build?  An example of the information I would want to know would be the limitation you just mentioned where BLE is available, but part of it doesn't work, but that it is targeted to be available at an estimated time.

**Reply by mradityagoyal** [reply]
Was this issue fixed?

**Comment by ZacharyInDallas** [comment]
Has this problem been fixed yet?

**Comment by MattyBoy4444** [comment]
Wondering when this will be fixed?

**Comment by andrewleech** [comment]
A community member has been working towards adding pairing and bonding support for PICO_W https://github.com/orgs/micropython/discussions/13612
